File: Sources/cpp/src/sprites/sprite.cpp
#include "sprite.h"
#include "timed_content_provider.h"

#include <chrono>
#include <iostream>
#include <string>
#include <vector>

Sprite::Sprite(const std::vector<std::string> frames, double fps) : timedContentProvider(frames, fps) {}

const std::string& Sprite::currentFrame() const {    
    return timedContentProvider.currentFrame();
}

void Sprite::update(std::chrono::milliseconds timeSinceLastUpdate) {
    timedContentProvider.update(timeSinceLastUpdate);
}

const uint32_t Sprite::numberOfFrames() const {
    return timedContentProvider.numberOfFrames();
}

File: Sources/windows/App/Sources/Sprites/Sprite.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace App
{
    public class Sprite
    {
        private TimedContentProvider<string> timedContentProvider;

        public Sprite(string animationName, List<string> frames, double fps)
        {
            this.AnimationName = animationName;
            this.timedContentProvider = new TimedContentProvider<string>(frames, fps);
        }

        public string AnimationName { get; private set; }

        public string CurrentFrame => timedContentProvider.CurrentFrame;

        public uint NumberOfFrames => (uint)timedContentProvider.NumberOfFrames;

        public void Update(TimeSpan timeSinceLastUpdate)
        {
            timedContentProvider.Update(timeSinceLastUpdate);
        }
    }
}
